### PR TITLE
Limit requested result set size to 1000 records

### PR DESCRIPTION
**Pull request by malcolmbaig**
Return an error to the client if the count param is higher than 1000.

Trello: https://trello.com/c/LbfXhokO

### CODE REVIEW

File: test/unit/parameter_parser/search_parameter_parser_test.rb [hunk 1]
```diff
@@ -168,6 +168,14 @@ def setup
     assert_equal(expected_params(count: 2), p.parsed_params)
   end
 
+  should "complain about an overly large count parameter" do
+    p = SearchParameterParser.new({ "count" => %w(1001) }, @schema)
+
+    assert_equal(%{Maximum result set size (as specified in 'count') is 1000}, p.error)
+    assert !p.valid?
```
Comment (tijmenb) [inline test/unit/parameter_parser/search_parameter_parser_test.rb]: `refute p.valid?`?

Comment (malcolmbaig) [inline test/unit/parameter_parser/search_parameter_parser_test.rb]: copy+paste from above, but yes, quite right.

**Comment by tijmenb** [comment]
👍 

Not a fan of line breaks in markdown documents btw, they make it harder to re-use the text elsewhere and make editing harder (if you add a long word you may need to re-linebreak whole paragraphs).

**Comment by benhyland** [comment]
👍 
looks fine, assuming further param validation (e.g. 0 <= start < count) is out of scope
although that example depends on what start is for...

File: test/unit/parameter_parser/search_parameter_parser_test.rb (hunk 1 above)
Comment (malcolmbaig) [inline test/unit/parameter_parser/search_parameter_parser_test.rb]: updated

**Comment by malcolmbaig** [comment]
@tijmenb point taken, but I'm not sure editing is all that easy when lines of text disappear off-screen. Text editors (once appropriately configured) are generally pretty decent at reflowing paragraphs of text to within 79 chars.

Also, regarding reuse - you could always copy/paste from the rendered markdown no? : )

**Pull request merged**